ERCS/erc-7720.md

```diff
@@ -0,0 +1,256 @@
+        uint256 unlockTime,
+        bytes32 referenceNo
+    );
+
+    // Event emitted when tokens are withdrawn.
+    event Withdraw(
+        uint256 txnId,
+        address indexed token,
+        address indexed from,
+        address indexed to,
+        uint256 amount
+    );
+
+    // Function to initiate a token transfer.
+    // Parameters:
+    // - _token: Address of the ERC20 token contract.
+    // - _from: Address of the sender.
+    // - _to: Address of the recipient.
+    // - _amount: Amount of tokens to be transferred.
+    // - _unlockTime: Time after which the tokens can be withdrawn.
+    // - _reference: Reference ID for the transaction.
+    // Returns the transaction ID.
+    function transferFrom(
+        address _token,
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint256 _unlockTime,
+        bytes32 _reference
+    ) external returns (uint256 txnId);
+
+    // Function to withdraw tokens from a transaction.
+    // Parameters:
+    // - _txnId: ID of the transaction to withdraw from.
+    function withdraw(uint256 _txnId) external;
+
+    // Function to get transaction details.
+    // Parameters:
+    // - _txnId: ID of the transaction.
+    // Returns the transaction details.
+    function getTransaction(uint256 _txnId)
+        external
+        view
+        returns (
+            address token,
+            address from,
+            address to,
+            uint256 amount,
+            uint256 unlockTime,
+            bytes32 referenceNo,
+            bool withdrawn
+        );
+}
+
+```
+
+## Rationale
+
+The design of the Deferred Token Transfer contract aims to provide a straightforward and secure method for handling time-locked token transfers. The following considerations were made during its development:
+
+**Simplicity and Usability**: The contract interface is designed to be simple and intuitive, making it easy for users to create deposits and for beneficiaries to withdraw tokens once the conditions are met.
+
+**Security**: The contract ensures secure token transfers, preventing common vulnerabilities associated with ERC-20 transfers. Additionally, the contract includes checks to prevent multiple withdrawals of the same deposit.
+
+**Flexibility**: The contract supports various ERC-20 tokens, allowing users to create deposits with any standard ERC-20 token. This flexibility makes it suitable for a wide range of use cases.
+
+**Event Logging**: Events are emitted for both deposit creation and token withdrawal. This provides transparency and allows easy tracking of contract activities, which is crucial for auditability and user confidence.
+
+**Conditional Payments**: By implementing a time-lock mechanism, the contract ensures that tokens are only transferred after a specific timestamp. This feature is essential for use cases like vesting schedules, escrow arrangements, and timed rewards, where payments need to be delayed until certain conditions are met.
```

Comment:
Your Rationale is pretty generic. These considerations kinda apply to all ERCs, and aren't specific to this one. Some examples you could explain here are why you chose a full `uint256` for `_unlockTime`, or why you've chosen to overload `transferFrom` instead of introducing a new function name.